sources/packages/backend/apps/api/src/route-controllers/education-program/_tests_/e2e/education-program.institutions.controller.createEducationProgram.e2e-spec.ts

```diff
@@ -166,6 +169,252 @@ describe("EducationProgramInstitutionsController(e2e)-createEducationProgram", (
         assessedById: null,
         submittedById: collegeFUser.id,
         isActive: true,
+        isExpired: false,
+      }),
+    );
+  });
+
+  it("Should throw duplicate SABC code for education program when there is already an active education program with the same SABC code.", async () => {
+    // Arrange
+    const sameSabcCode = `${faker.random.alpha({ count: 3 })}9`;
+    const existingEducationProgram = createFakeEducationProgram(
+      {
+        institution: collegeF,
+        user: collegeFUser,
+      },
+      {
+        initialValue: {
+          sabcCode: sameSabcCode,
+        } as Partial<EducationProgram>,
+      },
+    );
+    await db.educationProgram.save(existingEducationProgram);
+    const programStatus = ProgramStatus.Approved;
+    const payload = {
+      name: faker.lorem.words(5),
+      description: faker.lorem.words(5),
+      credentialType: "undergraduateCertificate",
+      cipCode: "11.1111",
+      fieldOfStudyCode: "15",
+      nocCode: "2174",
+      sabcCode: sameSabcCode,
+      institutionProgramCode: faker.random.alpha({ count: 3 }),
+      programIntensity: "Full Time and Part Time",
+      programDeliveryTypes: {
+        deliveredOnSite: true,
+        deliveredOnline: false,
+      },
+      completionYears: "12WeeksTo52Weeks",
+      courseLoadCalculation: "credit",
+      regulatoryBody: "other",
+      otherRegulatoryBody: "Other RB test",
+      entranceRequirements: {
+        minHighSchool: true,
+        hasMinimumAge: true,
+        requirementsByInstitution: true,
+        requirementsByBCITA: true,
+        noneOfTheAboveEntranceRequirements: false,
+      },
+      eslEligibility: "lessThan20",
+      hasJointInstitution: "no",
+      hasWILComponent: "no",
+      hasTravel: "no",
+      hasIntlExchange: "no",
+      programDeclaration: true,
+    };
```

Comment:
Is this entire setup the same for all the E2E tests being added?
If yes it can be centralized in a private method in this file.